scripts/build_artifact

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash -euo pipefail
```

Comment:
`-u` means "Attempt to use undefined variable outputs error message, and forces an exit" so this is how we can sort of gracefully error if `ARTIFACT_DIRECTORY` or `ARTIFACT_DESTINATION_FILE` are not defined when this is called

If this script gets more use or causes problems we could do some `if -z` checks and provide even clearer messages earlier on, but I think it's good enough for now

Reply:
is there an ok middle ground to have the Makefile call the script?

Reply:
I guess we could? I think what bugs me the most is that the destination of that task is required + dynamic, like if it was a static destination we could have a default and not require any args, but since it requires args, I think it makes for a confusing makefile step. With every single other step we can call `make x` and no arguments needed, but this one is different.

And this gets into how we use make also, I think make's special sauce is its use of target/destination directives for better dependency management (checking the mtime of dependency files, etc etc) and basically none of our commands use that, so maybe my issue is more on how we use make in general

Reply:
no problem, I'm fine leaving it out!